### PR TITLE
Update how_to_add_packages.md

### PR DESCRIPTION
**Pull request by prince-chrismc**
Docs!

➡️  Trying to improve this section since it's critical to new contributors and it was recently called out with some complaints.

- Restructured paragraphs
   - provides easy 1, 2, 3 at the start
   - broke out build service
   - removed duplication
- Syntax highlighting
- fixed wonky white spacing

I did my best to improve the english but it's my 2nd language so please let me know if you spot something!

As always, don't hesitate to provide feedback 🙏

### CODE REVIEW

File: docs/how_to_add_packages.md [hunk 1]
```diff
@@ -1,61 +1,61 @@
...
 requesting access in this GitHub [issue](https://github.com/conan-io/conan-center-index/issues/4). The EAP was designed to onboard authors
 to the new process.
 
 All EAP requests are reviewed and approved (or denied) every week, thus your request can take one week to be approved, so don't worry. This
 process helps conan-center-index against spam and malicious code.
 
-The contribution of packages is done via pull requests to the recipes found in this Github repository https://github.com/conan-io/conan-center-index.
+## Submitting a Package
+
+:two: To contribute a packages, you can submit a [Pull Request](https://github.com/conan-io/conan-center-index/pulls) to this GitHub repository https://github.com/conan-io/conan-center-index.
```
Comment (jgsogo) [inline docs/how_to_add_packages.md:34]:
```suggestion
:two: To contribute a package, you can submit a [Pull Request](https://github.com/conan-io/conan-center-index/pulls) to this GitHub repository https://github.com/conan-io/conan-center-index.
```

File: docs/how_to_add_packages.md [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
 
 The specific steps to add new packages are:
-* Fork the [conan-center-index](https://github.com/conan-io/conan-center-index) git repository, and then clone it.
-* Create a new folder with the Conan package recipe (conanfile.py).
-* Push to GitHub, and submit a pull request.
+* Fork the [conan-center-index](https://github.com/conan-io/conan-center-index) git repository, and then clone it locally.
+* Create a new folder with the Conan package recipe (conanfile.py) in the correct folder.
```
Comment (jgsogo) [inline docs/how_to_add_packages.md:38]:
```suggestion
* Create a new folder with the Conan package recipe (`conanfile.py`) in the correct folder.
```

File: docs/how_to_add_packages.md [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
 * Our automated build service will build 100+ different configurations, and provide messages that indicate if there were any issues found during the pull request on GitHub.
 
-When the pull request is reviewed and merged, those packages are published to [JFrog ConanCenter](https://conan.io/center/).
-
-## More Information about Recipes
-
-The [conan-center-index](https://github.com/conan-io/conan-center-index) (this repository) contains recipes for the remote [conan-center](https://conan.io/center/) repository. This remote is added by default to a clean installation of the Conan client.
-
-Recipes are contributed by opening pull-requests to this `conan-center-index` repository as it is explained in the section below.
+:three: When the pull request is [reviewed and merged](review_process.md), those packages are published to [JFrog ConanCenter](https://conan.io/center/).
```
Comment (jgsogo) [inline docs/how_to_add_packages.md:43]:
```suggestion
:three: When the pull request is [reviewed and merged](review_process.md), those packages are published to [JFrog ConanCenter](https://conan.io/center/) and available for everyone.
```

File: docs/how_to_add_packages.md [hunk 4]
```diff
@@ -75,13 +75,14 @@ e.g:
...
-### The version folder/s
+### The version folder(s)
 
 The system supports to use the same recipe for several versions of the library and also to create different recipes for different versions
 
 - **1 version => 1 recipe**
 
-  When the recipes change significantly between different library versions and reusing the recipe is not worth it, you can create a folder for each version and create inside both the “conanfile.py” and the “test_package” folder:
+  When the recipe changes significantly between different library versions and reusing the recipe is not worth it, you can create a folder for each new version and
+  create both inside the `conanfile.py` and the `test_package` in the version folder:
```
Comment (jgsogo) [inline docs/how_to_add_packages.md:85]:
```suggestion
  When the recipe changes significantly between different library versions and reusing the recipe is not worth it, it is recommended to create different folders and group together the
  versions that can share the same recipe. Each of these folders will require its own `conanfile.py` and `test_package` folder:
```

File: docs/how_to_add_packages.md [hunk 5]
```diff
@@ -96,7 +97,8 @@ The system supports to use the same recipe for several versions of the library a
 
 - **N versions => 1 recipe**
 
-   Create a folder named `all` (just a convention) and put both the “conanfile.py” and the “test_package” folder there. With this approach, the “conanfile.py” won’t declare the `version` attribute.
+   Create a folder named `all` (just a convention) and put both the `conanfile.py` and the `test_package` folder there. With this approach, the `conanfile.py`
+   won’t declare the `version` attribute.
```
Comment (jgsogo) [inline docs/how_to_add_packages.md:101]:
```suggestion
   Create a folder named `all` (just a convention) and put both the `conanfile.py` and the `test_package` folder there.
```

We can skip that part, we never add the `version` attribute, right?

**Review by jgsogo — COMMENTED** [review]
Some minor comments, probably a matter of style 👍

File: docs/how_to_add_packages.md (hunk 5 above)
Comment (prince-chrismc) [reply docs/how_to_add_packages.md:101]: That's an excellent question. I didn't fact check this it was already there

Comment (prince-chrismc) [reply docs/how_to_add_packages.md:101]: https://github.com/conan-io/conan-center-index/blob/6d1c2e29d7623af9654f614c7a7d1041a5e1d1db/recipes/zlib/1.2.11/conanfile.py#L8

I don't think this pattern is used at all

**Comment by prince-chrismc** [comment]
More advanced autotools #5889
No cmake namespace #5870

**Review by jgsogo — APPROVED** [review]
btw, there is no need to update manually the TOC, an automatic action will take care of them.

**Review by uilianries — APPROVED** [review]

**Comment by prince-chrismc** [comment]
> btw, there is no need to update manually the TOC, an automatic action will take care of them.

I know but it bothers me to approve a second PR for my own changes

**Review by tisonkun — APPROVED** [review]
Nice content.

One more comment is that in the case of https://github.com/conan-io/conan-center-index/pull/6028 it is not quite precise about "Add Package" but "Update Package". I did see this document but at the first time it seems irrelevant to my request.

**Comment by prince-chrismc** [comment]
Thanks for the feedback! I'll try to find a a way of bridging those two ideas in a follow up PR =)

**Comment by prince-chrismc** [comment]
👋 @madebr @SpaceIm @ericLemanissier @Croydon Looking for one last review for some doc changes 🙏

File: docs/how_to_add_packages.md [hunk 6]
```diff
@@ -149,12 +151,14 @@ If your sources are on GitHub, you can copy the link of the "Download ZIP" locat
...
 
-```
+```py
  def source(self):
-     tools.get(**self.conan_data["sources"][self.version])
+     tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
 ```
 
-### How to provide a good recipe
+> :warning: Please be aware that `conan-center-index` only builds from sources and pre-compiled binaries are not acceptable. For more information see our [packaging policy](https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md).
```
Comment (madebr) [inline docs/how_to_add_packages.md:159]: There are recipes that package binaries in this repo: msys2 and zulu-openjdk.
Perhaps this line should state that cci tries to build from source as much as possible and that packaging binaries is discouraged.

**Review by madebr — APPROVED** [review]

File: docs/how_to_add_packages.md (hunk 6 above)
Comment (prince-chrismc) [reply docs/how_to_add_packages.md:159]: #5792 is a recent example of this too

**Pull request merged**